x-pack/legacy/plugins/alerting/server/plugin.ts

```diff
@@ -0,0 +1,186 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import Hapi from 'hapi';
+import { Logger, Services } from './types';
+import { SpacesPlugin } from '../../spaces';
+import { ActionsPlugin } from '../../actions';
+import { AlertsClient } from './alerts_client';
+import { TaskManager } from '../../task_manager';
+import { AlertTypeRegistry } from './alert_type_registry';
+import { XPackMainPlugin } from '../../xpack_main/xpack_main';
+import { AlertsClientFactory } from './lib';
+import { KibanaRequest, SavedObjectsLegacyService } from '../../../../../src/core/server';
+import { EncryptedSavedObjectsPlugin } from '../../encrypted_saved_objects';
+import { ElasticsearchPlugin } from '../../../../../src/legacy/core_plugins/elasticsearch';
+import { PluginSetupContract as SecurityPluginSetupContract } from '../../../../plugins/security/server';
+import {
+  createAlertRoute,
+  deleteAlertRoute,
+  findAlertRoute,
+  getAlertRoute,
+  listAlertTypesRoute,
+  updateAlertRoute,
+  enableAlertRoute,
+  disableAlertRoute,
+  updateApiKeyRoute,
+  muteAllAlertRoute,
+  unmuteAllAlertRoute,
+  muteAlertInstanceRoute,
+  unmuteAlertInstanceRoute,
+} from './routes';
+
+interface PluginInitializerContext {
+  logger: {
+    get: () => Logger;
+  };
+}
+
+interface CoreSetup {
+  elasticsearch: ElasticsearchPlugin;
+  savedObjects: SavedObjectsLegacyService;
+  http: {
+    route: (route: Hapi.ServerRoute) => void;
+    basePath: {
+      serverBasePath: string;
+    };
+  };
+}
+
+interface PluginsSetup {
+  security?: SecurityPluginSetupContract;
+  task_manager: TaskManager;
+  spaces: () => SpacesPlugin | undefined;
+  actions: ActionsPlugin;
+  xpack_main: XPackMainPlugin;
+  encrypted_saved_objects: EncryptedSavedObjectsPlugin;
+}
+
+export interface PluginSetupContract {
+  registerType: AlertTypeRegistry['register'];
+}
+
+export interface PluginStartContract {
+  listTypes: AlertTypeRegistry['list'];
+  getAlertsClientWithRequest: (request: Hapi.Request) => AlertsClient;
+}
+
+export class Plugin {
+  private readonly logger: Logger;
+  private alertTypeRegistry?: AlertTypeRegistry;
+  private alertsClientFactory?: AlertsClientFactory;
+
+  constructor(initializerContext: PluginInitializerContext) {
+    this.logger = initializerContext.logger.get();
```

Comment:
If you switch to core's logger you'll have to provide the correct logger "context": `this.logger = initializerContext.logger.get('plugins', 'alerting');`

Reply:
Fixed in https://github.com/elastic/kibana/pull/46954/commits/68a6bcd25ec3a67e555d9111e554011abed678fe.